### PR TITLE
Add support for 128 byte token names

### PR DESCRIPTION
**Pull request by AStox**
Adds support for 128 byte token Names instead of only support 32byte token names and padding them to 128 bytes

### CODE REVIEW

File: src/gateway/Messages.sol [hunk 1]
```diff
@@ -820,6 +817,38 @@ library Messages {
     }
 
     // TODO: should be moved to a util contract
+    function stringToBytes128(string memory source) internal pure returns (bytes memory) {
```
Comment (offerijns) [inline src/gateway/Messages.sol:820]: I would move these and the bytes32 similar methods to a new `src/util/StringLib.sol`. We should also add some basic unit tests for these then.

File: src/gateway/Messages.sol [hunk 2]
```diff
@@ -846,6 +840,38 @@ library Messages {
     }
 
     // TODO: should be moved to a util contract
```
Comment (offerijns) [inline src/gateway/Messages.sol:842]:
```suggestion
```

File: src/gateway/Messages.sol [hunk 3]
```diff
@@ -846,6 +840,38 @@ library Messages {
...
+    function stringToBytes128(string memory source) internal pure returns (bytes memory) {
```
Comment (offerijns) [inline src/gateway/Messages.sol:843]: Let's adopt the _ pattern for internal functions
```suggestion
    function _stringToBytes128(string memory source) internal pure returns (bytes memory) {
```

File: src/gateway/Messages.sol [hunk 4]
```diff
@@ -846,6 +840,38 @@ library Messages {
...
+                result[i] = temp[i];
+            } else {
+                result[i] = 0x00;
+            }
+        }
+
+        return result;
+    }
+
+    function bytes128ToString(bytes memory _bytes128) internal pure returns (string memory) {
```
Comment (offerijns) [inline src/gateway/Messages.sol:858]:
```suggestion
    function _bytes128ToString(bytes memory _bytes128) internal pure returns (string memory) {
```

File: src/gateway/Messages.sol [hunk 5]
```diff
@@ -846,6 +840,38 @@ library Messages {
...
+        bytes memory bytesArray = new bytes(i);
+
+        for (uint8 j = 0; j < i; j++) {
+            bytesArray[j] = _bytes128[j];
+        }
+
+        return string(bytesArray);
+    }
+
     function stringToBytes32(string memory source) internal pure returns (bytes32 result) {
```
Comment (offerijns) [inline src/gateway/Messages.sol:875]:
```suggestion
    function _stringToBytes32(string memory source) internal pure returns (bytes32 result) {
```

File: src/gateway/Messages.sol (hunk 3 above)
Comment (AStox) [reply src/gateway/Messages.sol:843]: Want me to update the internal functions in TestSetup.t.sol too?

Comment (offerijns) [reply src/gateway/Messages.sol]: Yes please do!

**Review by offerijns — APPROVED** [review]

**Pull request merged**